src/synapse/azext_synapse/_validators.py

```diff
@@ -4,18 +4,26 @@
 # --------------------------------------------------------------------------------------------
 
 
-def example_name_or_id_validator(cmd, namespace):
-    #  pylint: disable=line-too-long
-    # Example of a storage account name or ID validator.
-    # See: https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#supporting-name-or-id-parameters
-    from azure.cli.core.commands.client_factory import get_subscription_id
-    from msrestazure.tools import is_valid_resource_id, resource_id
-    if namespace.storage_account:
-        if not is_valid_resource_id(namespace.RESOURCE):
-            namespace.storage_account = resource_id(
-                subscription=get_subscription_id(cmd.cli_ctx),
-                resource_group=namespace.resource_group_name,
-                namespace='Microsoft.Storage',
-                type='storageAccounts',
-                name=namespace.storage_account
-            )
+from msrestazure.tools import is_valid_resource_id
+from azure.cli.core.commands.parameters import get_resources_in_subscription  # pylint: disable=unused-import
+
+
+def validate_storage_account(namespace):
+    from msrestazure.tools import parse_resource_id
+    if is_valid_resource_id(namespace.storage_account):
+        parsed_workspace = parse_resource_id(namespace.storage_account)
+        storage_name = parsed_workspace['resource_name']
+        namespace.storage_account = storage_name
+
+
+def validate_statement_language(namespace):
+    statement_language = {
+        'spark': 'spark',
+        'scala': 'spark',
+        'pypark': 'pyspark',
+        'python': 'pyspark',
+        'sparkdotnet': 'sparkdotnet',
+        'csharp': 'sparkdotnet',
+        'sql': 'sql'
+    }
+    namespace.language = statement_language.get(namespace.language.lower())
```

Comment:
What will happen if users pass some invalid language?

Reply:
This line `c.argument('language', arg_type=get_enum_type(SparkStatementLanguage))` requires it must be one of SparkStatementLanguage. It will check automaticlly and return errors just like bellow if it is invalid:

"az synapse spark statement invoke: 'fwef' is not a valid value for '--language'. See 'az synapse spark statement invoke --help'."


---
In reply to: [421374967](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r421374967) [](ancestors = 421374967)